docs/prefetch.md

```diff
@@ -0,0 +1,112 @@
+# How does prefetch work
+
+## Render Runtime related
+
+The prefetch in render-runtime has a few key concepts:
+- The prefetch hook, that is used by the `<Link>` component, which triggers the prefetch at the appropriate moment
+- The prefetch context, that holds the state with a bunch o objects that are important for the prefetch logic
+- On RenderProvider when a navigation is triggered, we go to the prefetch cache to see if there is data there.
+
+### The Prefetch context
+
+This is the prefetch state interface:
+```
+
+interface PrefetchState {
+  routesCache: LRUCache<PrefetchRouteData>
+  pathsCache: {
+    other: LRUCache<PrefetchCacheObject>
+    product: LRUCache<PrefetchCacheObject>
+    search: LRUCache<PrefetchCacheObject>
+  }
+  pathsState: Record<string, PathState>
+  routePromise: Record<string, RoutePromise | null>
+  queue: PQueue
+}
+```
```

Comment:
to have the highlight of the typescript in the markdown
```suggestion

```tsx
interface PrefetchState {
  routesCache: LRUCache<PrefetchRouteData>
  pathsCache: {
    other: LRUCache<PrefetchCacheObject>
    product: LRUCache<PrefetchCacheObject>
    search: LRUCache<PrefetchCacheObject>
  }
  pathsState: Record<string, PathState>
  routePromise: Record<string, RoutePromise | null>
  queue: PQueue
}
```
```